router/h2/src/main/scala/io/buoyant/router/h2/ClassifierFilter.scala

```diff
@@ -135,9 +135,7 @@ class ClassifierFilter(classifier: H2Classifier) extends SimpleFilter[Request, R
                 // replace it with a new data frame without that flag.
                 // otherwise, the trailers we add to the stream will never
                 // be reached.
-                val frame2 = Frame.Data(frame.buf, eos = false)
-                // release the old data frame, since it will be replaced.
-                frame.release()
```

Comment:
As we discussed previously, releasing the frame here was almost certainly incorrect. This LGTM 👍